src/utils.js

```diff
@@ -26,6 +26,27 @@ const fetchArticle = ({ article_id }) => {
     });
 };
 
+const patchVote = (article_id) => {
+    return newsApi
+    .patch(`/articles/${article_id}`,{inc_votes:1})
+}
+
+const updateVote = (value, voteCount, setVoteCount, article_id, setDisabled, setHideVoteFail) => {
+    let updatedVoteCount = voteCount + value
+    setVoteCount(updatedVoteCount);
+    patchVote(article_id)
+      .then(() => {
+        setDisabled(true)
+        setHideVoteFail(true)
+      })
+      .catch((err) => {
+        setVoteCount(updatedVoteCount - value);
+        setDisabled(false)
+        setHideVoteFail(false)
```

Comment:
Nice addition of error handling to make sure the user knows if the vote was unsuccessful 👍